src/services/usersService.ts

```diff
@@ -0,0 +1,109 @@
+import { Disposable } from "vscode";
+import { AltimateRequest } from "../altimate";
+import { DBTTerminal } from "../dbt_client/dbtTerminal";
+import { provideSingleton } from "../utils";
+import { DBTProjectContainer } from "../manifest/dbtProjectContainer";
+import { DBTInstallationVerificationEvent } from "../dbt_client/dbtVersionEvent";
+
+export interface TenantUser {
+  id: number;
+  uuid: string;
+  display_name: string;
+  first_name: string;
+  last_name: string;
+  email: string;
+  phone: string;
+  is_active: boolean;
+  is_verified: boolean;
+  is_invited: boolean;
+  is_onboarded: boolean;
+  created_at: string;
+  role_title: string;
+}
+
+/**
+ * Service to load and store users in tenant and current user
+ */
+@provideSingleton(UsersService)
+export class UsersService implements Disposable {
+  // Local cache of users in tenant
+  private tenantUsers: Record<string, TenantUser> = {};
+  // Local cache of current user
+  private tenantUser: TenantUser | undefined;
+  private disposables: Disposable[] = [];
+
+  public constructor(
+    private dbtProjectContainer: DBTProjectContainer,
+    private dbtTerminal: DBTTerminal,
+    private altimateRequest: AltimateRequest,
+  ) {
+    this.disposables.push(
+      this.dbtProjectContainer.onDBTInstallationVerification((e) =>
+        this.onDBTInstallationVerification(e),
+      ),
+    );
+  }
+
+  dispose() {
+    while (this.disposables.length) {
+      const x = this.disposables.pop();
+      if (x) {
+        x.dispose();
+      }
+    }
+  }
+
+  private onDBTInstallationVerification(
+    event: DBTInstallationVerificationEvent,
+  ) {
+    if (event.installed) {
+      this.loadCurrentUser();
+      this.loadUsersInTenant();
+    }
+  }
+
+  private async loadUsersInTenant() {
+    if (this.altimateRequest.getCredentialsMessage()) {
+      this.dbtTerminal.debug(
+        "UsersService:loadUsersInTenant",
+        "Missing credentials. skipping loadUsersInTenant",
+      );
+      return;
+    }
+    this.dbtTerminal.debug("UsersService", "loading tenant users");
+    const users = await this.altimateRequest.fetch<TenantUser[]>("/users/chat");
```

Comment:
Should not call fetch directly but create facade method in altimate.ts.

Reply:
fixed

Reply:
There are many more instances of where we don't do this. Either we stop doing this as long as it is consistent.